File: IKeep.Web/ClientApp/src/services/dtos/NewChoresRequest.js
class NewChoresRequest
{
    constructor(json)
    {
        if(json)
        {
            this.InstallationId = json.installationId;
            
            this.BuildingId = json.buildinId;
            this.FloorId = json.floorId;
            this.AreaId = json.areaId;

            this.Year = json.year;
            this.Month = json.month;
            this.Day = json.day;
        }
        else
        {
            this.InstallationId = "";
            this.BuildingId = "";
            this.FloorId = "";
            this.AreaId = "";
            this.Year = "";
            this.Month = "";
            this.Day = "";
        }
    }
}

export default NewChoresRequest;

File: IKeep.Lib/Services/Dtos/NewChoresRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace IKeep.Lib.Services.Dtos
{
    public class NewChoresRequest
    {
        public Guid? InstallationId { get; set; }
        public Guid? BuildingId { get; set; }
        public Guid? FloorId { get; set; }
        public Guid? AreaId { get; set; }
        public int Year { get; set; }
        public int? Month { get; set; }
        public int? Day { get; set; }
        public bool AllInstallations { get; set; }

    }
}
